File: src/main/java/com/tobe/healthy/ApiResult.java
package com.tobe.healthy;

import org.springframework.http.HttpStatus;

import lombok.Getter;

@Getter
public class ApiResult<T> {

	private final HttpStatus status;
	private final String message;
	private final T data;

	public ApiResult(String message, T data) {
		this.status = HttpStatus.OK;
		this.message = message;
		this.data = data;
	}

	public static <T> ApiResult<T> success(String message, T data) {
		return new ApiResult<>(message, data);
	}

	public static ApiResult<Void> success(String message) {
		return new ApiResult<>(message, null);
	}
}

File: src/main/kotlin/com/tobe/healthy/ApiResult.kt
package com.tobe.healthy

import mu.KotlinLogging
import org.springframework.http.HttpStatus
import org.springframework.http.HttpStatus.OK

val log = KotlinLogging.logger { }

data class ApiResult<T>(
    val status: HttpStatus = OK,
    val message: String,
    val data: T
)
